### DISCUSSION TITLE
MongoDB GridFS support?  Inserts with BulkMode?  What do YOU want or need?

### DISCUSSION THREAD

**Discussion by thadguidry** [9tigerio/db2rest#756] [Ideas]
Wondering if our users have any need for:

- add [GridFS support](https://www.mongodb.com/docs/manual/core/gridfs/?msockid=0089d04b600a6b453186c27c61496ac9) for MongoDB for storing and querying binary files (> 16 MB) and searching and filtering on any objects within a files' metadata field objects.  This would be useful for folks that might be doing things with photos, videos, huge binary files, machine learning, etc.  The cool thing is that when files are uploaded, GridFS automatically creates indexes on the `files` and `chunks` collections when you upload data to the GridFS bucket.  But we'd also provide easy automatic REST API that can create Compound Indexes `index_A(field1,field2)` and perhaps others like Multikey indexes over array-valued fields, or Geospatial indexes over GeoJSON objects. 

- [Time Series Data](https://www.mongodb.com/docs/languages/java/reactive-streams-driver/current/data-formats/time-series/) - ( Location, sensor type, etc.) (Monthly rent price, Location, currency) - Any metadata that describes a measurement - We would support Inserting One or Many Documents into Collections with a `timestamp` field and allowing read or aggregation operations as well as perhaps setting [custom bucketing granularity](https://www.mongodb.com/docs/manual/core/timeseries/timeseries-granularity/) on a collection.

- there's also quite a bit of MongoDB stuff that we'd like to add support for like [Bulk Insert Mode]( https://www.mongodb.com/docs/languages/java/reactive-streams-driver/current/write/bulk-writes/) and any other's that [Spring Data](https://docs.spring.io/spring-data/mongodb/reference/mongodb.html) allows easy support for to make it easy for contributors to wire up that support in Java and submit PR's to us.

- and probably others we haven't thought much more about, but let us know what your direct needs are with MongoDB support!

**Comment by kdhrubo** [comment]
MongoDB support to be dropped. So far there has been not many requests or tickets or issues for Mongo. Seems the usage is not high to justify development. So in v 2.0 mongoDB and Oracle 9i support will be dropped around April 2025.
